SpockOpaleye/src/TenantApi.hs

```diff
@@ -58,50 +40,25 @@ deactivate_tenant :: Connection -> Tenant -> IO Tenant
 deactivate_tenant conn tenant = set_tenant_status conn tenant TenantStatusInActive
 
 set_tenant_status :: Connection -> Tenant -> TenantStatus -> IO Tenant
-set_tenant_status conn tenant status = update_tenant conn (tenant_id tenant)
-                                              tenant { tenant_status = status }
+set_tenant_status conn tenant st = update_tenant conn (tenant ^. id) (tenant & status .~ st)
 
 update_tenant :: Connection -> TenantId -> Tenant -> IO Tenant
```

Comment:
The Opaleye files were split to resolve a circular imports issue that I came across. But it does not seem to be an issue any more and I have merged them back.

Will convert code from snake_case to camelCase.

>What is the advantage of passing the id separately?

Passing the id separately explicitly specify which record to update. The record contains the data that the row should be updated with. The id specify the row that should be updated. Since those are separate things, I think those must remain separate.

Imagine a case when someone have a record where they have made some updates to the fields, and want to save to a certain Id. But imagine that they missed to change the id field. When they make the call, the wrong record is going to get updated. There is nothing warning the user that they didn't specify the id of the row that should be updated, because that was implicitly assumed to be the id that came with the updated record.